egs/wsj/s5/steps/nnet3/get_egs.sh

```diff
@@ -265,23 +293,70 @@ valid_egs_opts="--left-context=$valid_left_context --right-context=$valid_right_
 
 echo $left_context > $dir/info/left_context
 echo $right_context > $dir/info/right_context
-num_pdfs=$(tree-info --print-args=false $alidir/tree | grep num-pdfs | awk '{print $2}')
-if [ $stage -le 3 ]; then
-  echo "$0: Getting validation and training subset examples."
-  rm $dir/.error 2>/dev/null
-  echo "$0: ... extracting validation and training-subset alignments."
 
+if ! $raw_nnet; then
```

Comment:
I reason I would be fine with adding the changes to the same python script is due to the extensive use of functions in python script. It is usually easier to read. If you do feel the python script is also getting complicated due to excessive branching you could consider splitting the train.py script. Its up to you how you want to do this.